<launch>
    <arg name="rviz" default="true"/>
    <arg name="enable_slam" default="true"/>
    <arg name="kalman_dead_reckoning" default="false"/>
    <arg name="number_of_robots" default="3"/>

    <!-- online mode -->
    <group ns="bruce">
        <group ns="slam">

            <!-- start the gyro intergration node if we are not using the kalman filer -->
            <node if="$(eval kalman_dead_reckoning == false)"  pkg="bruce_slam" name="gyro_fusion" type="gyro_node.py" output="screen">
                <rosparam command="load" file="$(find bruce_slam)/config/gyro.yaml"/>
            </node>

            <!-- start the dead reckoning node if we are not using the kalman filter -->
            <node if="$(eval kalman_dead_reckoning == false)"  pkg="bruce_slam" name="dead_reckoning" type="dead_reckoning_node.py" output="screen">
                <rosparam command="load" file="$(find bruce_slam)/config/dead_reckoning.yaml"/>
            </node>

            <!-- start the kalman filter if we have requested it above -->
            <node if="$(eval kalman_dead_reckoning == true)"  pkg="bruce_slam" name="kalman" type="kalman_node.py" output="screen">
                <rosparam command="load" file="$(find bruce_slam)/config/kalman.yaml"/>
            </node>
            
            <!-- start the feature extraction node -->
            <node pkg="bruce_slam" name="feature_extraction" type="feature_extraction_node.py" output="screen">
                <rosparam command="load" file="$(find bruce_slam)/config/feature.yaml" subst_value="true"/>
            </node>

            <!-- start the SLAM node -->
            <node pkg="draco_slam" name="slam" type="multi_robot_slam_node.py" output="screen">
                <rosparam command="load" file="$(find bruce_slam)/config/slam.yaml" subst_value="true"/>
                <rosparam command="load" file="$(find draco_slam)/config/multi_robot.yaml" subst_value="true"/>
                <param name="enable_slam" value="$(arg enable_slam)"/>
                <param name="save_fig" value="false"/>
                <param name="rov_number" value="rov_one"/>
                <param name="number_of_robots" value="$(arg number_of_robots)"/>
            </node>

        </group>
        <node pkg="tf" name="map_to_world_tf_publisher" type="static_transform_publisher" args="0 0 0 0 0 3.14159 world map 100"/>
    </group>

    <node if="$(arg rviz)" type="rviz" name="rviz" pkg="rviz" args="-d $(find bruce_slam)/rviz/video.rviz" />

</launch>
